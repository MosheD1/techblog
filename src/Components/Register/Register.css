main {
    margin: 4rem;
}   

.container {
    margin: auto;
    width: 600px;

}

.register {
    padding: 20px;
    outline: none;
    border: 1px solid #ccc;
    box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.4);
    border-radius: 10px;
}

fieldset {
    border: none;
}

input[type=submit] {
    display: block;
    margin: 0.5rem auto;
    padding: 0.8rem;
    background-color: lightgreen;
    border-radius: 4px;
    outline: none;
    border: none;
}