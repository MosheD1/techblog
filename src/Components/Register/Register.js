import React, {Component} from 'react'
import './Register.css'

class Register extends Component {
    constructor(props) {
        super(props);
        this.state = {
            registerName: '',
            registerEmail: '',
            registerPassword: ''
        }
    }

    onNameChange = (e) => {
        this.setState({registerName: e.target.value});
    }

    onEmailChange = (e) => {
        this.setState({registerEmail: e.target.value});
    }

    onPasswordChange = (e) => {
        this.setState({registerPassword: e.target.value});
    }

    render() {
        return(
            <main>
                <div className='container register'>
                    <fieldset>
                        <legend>Register</legend>
                        <div className='widget'>
                            <label>Name</label>
                            <input onChange={this.onNameChange} type='text'></input>
                        </div>
                        <div className='widget'>
                            <label>Email</label>
                            <input onChange={this.onEmailChange} type='text'></input>
                        </div>
                        <div className='widget'>
                            <label>Password</label>
                            <input onChange={this.state.onPasswordChange} type='text'></input>
                        </div>
                    </fieldset>
                    <input onClick={() => this.props.onRouteChange('home')} type='submit' value='Register'></input>
                </div>
        </main>
        );
    }
}

export default Register;