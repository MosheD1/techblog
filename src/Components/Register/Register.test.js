import React from 'react';
import Register from './Register';
import {shallow, render, mount} from 'enzyme';

let mockState = {
    registerName: '',
    registerEmail: '',
    registerPassword: ''
}

describe('Register', () => {
    it('expect Register to be render', () => {
        expect(shallow(<Register />)).toMatchSnapshot();
    });

    it('expect to get initial state', () => {
        expect(shallow(<Register />).state()).toEqual(mockState);
    });
});