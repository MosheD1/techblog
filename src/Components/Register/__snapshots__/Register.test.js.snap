// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register expect Register to be render 1`] = `ShallowWrapper {}`;
